pkg/virt-handler/vm.go

```diff
@@ -508,7 +508,7 @@ func (d *VirtualMachineController) hasGracePeriodExpired(dom *api.Domain) (hasEx
 	hasExpired = false
 	timeLeft = 0
 
-	if dom == nil {
+	if dom == nil || dom.Spec.Metadata.KubeVirt.GracePeriod == nil {
```

Comment:
`dom.Spec.Metadata.KubeVirt.GracePeriod` when is this a `nil`?

Reply:
@nekkunti cc

Reply:
when  GracePeriod not set in vmi , then this will be nil

Reply:
As far as I see in the code the GracePeriod is always set, can you point me to code or some crash?

Reply:
The only case this can happen when we have a domain which is not populated from the informer but rather inserted into cache directly. In this case the absence of the `GracePeriod` should not mean that we have hit the grace period.

Reply:
@nekkunti ptal

Reply:
@xpivarc 

I don’t recall the exact scenario, but I believe one of the end-to-end (e2e) tests or unit tests was returning nil for GracePeriod, which was causing a crash. To prevent this, I added the nil check


However, I got ur point  but i think it is safe to assume  GracePeriod expired if it is nil due to any reason at this point. 
If you believe removing this condition would be more appropriate, I can remove the GracePeriod == nil check and trigger a new build to see if the tests pass without issues( I feel it not adding much value here  to debug)

Please let me know how you’d like to proceed.

Reply:
@xpivarc  ^^

Reply:
Let's try to remove this and run unit and e2e. Unfortunately it is not safe assumption.

Reply:
k . let me try